### PR TITLE
Frontend: Use GeoContext for population counts and SVI

### PR DESCRIPTION
**Pull request by benhammondmusic**
## Description

- refactor PopulationCard to make single query to GeoContext for total population and SVI (which is only used at county-level)
- updates MapCard with simpler SVI fetch
- refactor DatasetOrganizer to sort on the "fips" column if there is no demographic column to sort on
- AcsProvider and Acs2010Provider are now only used on the remaining topics that are calculated on the frontend. These can probably be deleted once #1474 is done
- refactors the CdcSviProvider into the GeoContextProvider; doesn't use a dummy demographic breakdown as we were before and can simply read the totals without filtering for the "All" rows
- adds GeoContextProvider tests

## Preview Link

<a href="https://deploy-preview-1957--health-equity-tracker.netlify.app/exploredata" target="_blank">
  <br />
Click to demo updates </a> 

## Motivation and Context

- Frontend for #1953 
- Making a single call to a tiny dataset rather than 2 calls at the county level with the population datasets being much larger

## How has this been tested?

manually confirm smaller / fewer fetches
Screen Shot 2023-01-10 at 9 54 46 AM
Screen Shot 2023-01-10 at 9 57 17 AM

- [x] adds tests for geo context provider

## Screenshots (if appropriate):

Population Card working as expected
Screen Shot 2023-01-09 at 11 30 34 PM
Screen Shot 2023-01-09 at 11 30 44 PM
Screen Shot 2023-01-09 at 11 30 57 PM
Screen Shot 2023-01-09 at 11 31 11 PM
Screen Shot 2023-01-09 at 11 31 19 PM
Screen Shot 2023-01-09 at 11 31 32 PM
Screen Shot 2023-01-09 at 11 31 45 PM
Note above in regards to #1954 

Map Card Hover SVI working as expected
Screen Shot 2023-01-09 at 11 48 54 PM
Screen Shot 2023-01-09 at 11 49 00 PM
Screen Shot 2023-01-09 at 11 49 23 PM
Screen Shot 2023-01-09 at 11 49 35 PM
Screen Shot 2023-01-09 at 11 49 59 PM

## Types of changes

- Refactor (code improvement that doesn't affect UI)

### CODE REVIEW

**Review by eriwarr — APPROVED** [review]
Looks great!

**Review by joshzarrabi — APPROVED** [review]
Very nice

File: frontend/src/data/variables/Acs2010PopulationProvider.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ class Acs2010PopulationProvider extends VariableProvider {
     );
     let acs2010DataFrame = acs2010Dataset.toDataFrame();
 
-    // If requested, filter geography by state or coacs2010ty level
```
[the remark this answers is not in the document]
Comment (benhammondmusic) [reply frontend/src/data/variables/Acs2010PopulationProvider.ts:55]: haha . i didn't git blame it but was curious where that came from

File: frontend/src/data/variables/GeoContextProvider.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    const geographyToACSDatasetMap: Record<GeographicBreakdown, string> = {
+      county: isAcs2010
+        ? "acs_2010_population-by_age_territory"
+        : "acs_population-by_age_county",
+      state: isAcs2010
+        ? "acs_2010_population-by_age_territory"
+        : "acs_population-by_age_state",
+      national: "acs_population-by_age_national",
+      // next entries are unused
```
[the remark this answers is not in the document]
Comment (benhammondmusic) [reply frontend/src/data/variables/GeoContextProvider.ts:43]: Just so we can use a `Record<`; typescript yells at you if you don't have every possible item accounted for. We could make it a `Partial<Record<` but i think you lose some of the nice autocomplete / autocompletion stuff

**Pull request merged**